syntax = "proto3";

package messages;

service serverQueries {
    //rpc getEntireHistory(Void)              returns();
    rpc getHistoryOfAllParticipants(Void)   returns(entireHistory);
}

message Void { }

message entireHistory {
    repeated participantHistory participantHistory = 1;
}

message participantHistory {
    participant participant = 1;
    repeated participantInformation participantInformation = 2;
}

message participant {
    string type = 1;
    uint32 id = 2;
}

message participantInformation {
    int32 value = 1;
    uint64 timestamp = 2;
}